src/typechecker.erl

```diff
@@ -3664,12 +3778,23 @@ add_type_pat({bin, _P, BinElements} = Bin, Ty, TEnv, VEnv, Caps) ->
                     BinElements),
     {type(none), BinTy, NewVEnv, Cs};
 add_type_pat({record, P, Record, Fields}, Ty, TEnv, VEnv, Caps) ->
-    case expect_record_type(Record, Ty, TEnv) of
-        type_error -> throw({type_error, record_pattern, P, Record, Ty});
-        {ok, Cs1} ->
-            {VEnv2, Cs2} = add_type_pat_fields(Fields, Record, TEnv, VEnv, Caps),
-            RecTy = {type, erl_anno:new(0), record, [{atom, erl_anno:new(0), Record}]},
-            {type(none), RecTy, VEnv2, constraints:combine(Cs1, Cs2)}
+    case expect_record_type(Ty, Record, TEnv) of
+        any ->
+            %% FIXME: If record() is used and we try to bind, every field types
+            %% - should be any(). add_any_types_pat won't work, will need a
+            %% - add_any_type_pat_fields
```

Comment:
I think it'd be good to get this fixed before merging.

Reply:
For sure, I just wanted to create the PR for early review, there is still a bit of work that needs to be done.

Reply:
I think it was actually right the way I wrote it. Otherwise it makes other stuff fail. Resolved for now.